ethpm_cli/auth.py

```diff
@@ -0,0 +1,40 @@
+import os
+from typing import Optional
+
+from eth_account import Account
+import eth_keyfile
+from eth_typing import Address
+from eth_utils import to_bytes
+
+from ethpm_cli.constants import KEYFILE_PASSWORD, KEYFILE_PATH
+from ethpm_cli.exceptions import ValidationError
+
+
+def set_auth(keyfile_path: str, password: str) -> None:
+    validate_keyfile_path_and_password(keyfile_path, to_bytes(text=password))
+    os.environ[KEYFILE_PATH] = keyfile_path
+    os.environ[KEYFILE_PASSWORD] = password
```

Comment:
This gives me pause.  Is there a reason you need to set these in the environment.  IIUC the motivation for this is so that you can set these up using a CLI command and then use them later?

I think this might be a problematic approach due to it exposing the key in the keyfile in a manner that is *effectively* storing it in plain text.  My suggestion for how we might do this in a manner that preserves security but still gives good UX.

1. Split this into two mechanisms.
    - A mechanism for recording the location of the keyfile that a user wishes to use for future commands.  I'd recommend using the filesystem and maybe a config file located somewhere like [`$XDG_CONFIG_HOME/ethpm_cli/`](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)
        - maybe a separate mechanism that allows *importing* a keyfile which actually copies the file into `XDG_CONFIG_HOME/ethpm_cli/keystore` for easy future use?
    - A mechanism for the **user** to provide the password via multiple mechanisms.
        - CLI prompt where they type it in
        - CLI flag where they provide it in plaintext
        - environment variable where they set it in `os.environ`